app/schema.json

```diff
@@ -42,11 +42,92 @@
         "initialYDomain": { "$ref": "#/definitions/domain" },
         "autocompleteSource": { "type": "string" },
         "genomePositionSearchBoxVisible": { "type": "boolean" },
-        "genomePositionSearchBox": { "type": "object", "TODO": "more precise" },
+        "genomePositionSearchBox": { "$ref": "#/definitions/genomePositionSearchBox" },
         "chromInfoPath": { "type": "string" },
-        "tracks": { "type": "object", "TODO": "more precise" },
-        "layout": { "type": "object", "TODO": "more precise" },
-        "overlays": { "type": "array" }
+        "tracks": { "$ref": "#/definitions/tracks_object" },
+        "layout": { "$ref": "#/definitions/layout" },
+        "overlays": { "type": "array", "items": { "$ref": "#/definitions/overlay" } }
+      }
+    },
+
+    "genomePositionSearchBox": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["autocompleteServer", "autocompleteId", "chromInfoServer", "chromInfoId", "visible"],
+      "properties": {
+        "autocompleteServer": { "type": "string" },
+        "autocompleteId": { "type": "string" },
+        "chromInfoServer": { "type": "string" },
+        "chromInfoId": { "type": "string" },
+        "visible": { "type": "boolean" }
+      }
+    },
+
+    "overlay": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "uid": { "type": "string" },
+        "type": { "type": "string" },
+        "includes": { "type": "array" },
+        "options": { "type": "object" },
+        "chromInfoPath": { "type": "string" }
+      }
+    },
+
+    "layout": {
+      "type": "object",
+      "additionalProperties": false,
+      "required": ["w", "h", "x", "y", "moved", "static"],
+      "properties": {
+        "w": { "type": "number" },
+        "h": { "type": "number" },
+        "x": { "type": "number" },
+        "y": { "type": "number" },
+        "i": { "type": "string" },
+        "moved": { "type": "boolean" },
+        "static": { "type": "boolean" }
+      }
+    },
+
+    "tracks_object": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "left": { "type": "array" },
+        "right": { "type": "array" },
+        "top": { "type": "array" },
+        "bottom": { "type": "array" },
+        "center": { "type": "array" },
+        "whole": { "type": "array" },
+        "gallery": { "type": "array" }
+      }
+    },
+
+    "tracks_array": {
+      "note": "I don't think we can enforce this because every track type has a different model?",
```

Comment:
Yeah, this one is tricky. There's a lot of variation in terms of how a track can be defined. This looks like a good start though.

Reply:
Would you like this much to be enforced? This would pass with all the examples that are checked in, but it's not enforced: In the section above, we only say that `left`, `right`, etc are arrays: 
```
"tracks_object": {
      "type": "object",
      "additionalProperties": false,
      "properties": {
        "left": { "type": "array" },
        "right": { "type": "array" },
        "top": { "type": "array" },
        "bottom": { "type": "array" },
        "center": { "type": "array" },
        "whole": { "type": "array" },
        "gallery": { "type": "array" }
      }
    },
```
Options:
- Status quo
- Validate against it. ( `"type": "array"` -> `"$ref": "#/definitions/tracks_array"` )
- Even stricter: We could check the track types, and make sure that those tracks have the particular properties that belong.